### PR TITLE
Modified ARM server_create code along with corresponding RSpecs to override image sku.

### PR DESCRIPTION
**Pull request by Aliasgar16**
Done.

### CODE REVIEW

**Comment by adamedx** [comment]
Does the Azure cli support RHEL? If RHEL is supported, we should add that as an os type.

**Comment by adamedx** [comment]
So I like override behavior with sku -- it overrides what we set when `os-image-type` is specified. However, we still need to let people specify the other parameters in a non-override format. So we should still allow the use of `Publisher`, `Offer`, and `Version`. The way this should work is the following:

`knife azurerm server create <os-image-type ostype [--image-sku sku] | image-publisher publisher image-offer offer --image-sku sku image-sku sku [--image-version version]>`

So you can specify the image in one of two main ways:

1. Mandatory specification of `os-image-type` and optionally override just the `sku`
2. Or mandatory specification of `publisher`, `offer` and `sku`, and optional specification of `image-version` with default of `latest`

**Comment by Aliasgar16** [comment]
Made changes to support all 4 image reference parameters along with os_type.

File: lib/chef/knife/azurerm_server_create.rb [hunk 1]
```diff
@@ -270,27 +270,27 @@ def ssh_override_winrm
...
         begin
           if locate_config_value(:azure_image_os_type)
-            if (locate_config_value(:azure_image_reference_publisher) || locate_config_value(:azure_image_reference_offer) || locate_config_value(:azure_image_reference_sku))
-              # if azure_image_os_type is given and other image reference parameters are also given,
+            if (locate_config_value(:azure_image_reference_publisher) || locate_config_value(:azure_image_reference_offer))
+              # if azure_image_os_type is given and any of the other image reference parameters like publisher or offer are also given,
               # raise error
-              raise ArgumentError, 'Please specify either --azure-image-os-type OR other image reference parameters i.e.
+              raise ArgumentError, 'Please specify either --azure-image-os-type OR --azure-image-os-type with --azure-image-reference-sku or 4 image reference parameters i.e.
                 --azure-image-reference-publisher, --azure-image-reference-offer, --azure-image-reference-sku, --azure-image-reference-version."'
```
Comment (NimishaS) [inline lib/chef/knife/azurerm_server_create.rb]: @adamedx , any suggestions on this error message? 
I think we should either simplify the error message OR just keep it the same way as it was `Please specify either --azure-image-os-type OR other image reference parameters` and explain the usage in the Readme.

**Pull request merged**